src/scraper/http.cpp

```diff
@@ -138,6 +138,49 @@ namespace
     };
 
 #endif
+
+    //!
+    //! \brief Parse an etag value from an HTTP header field.
+    //!
+    //! This will parse etag values from headers in these standard formats:
+    //!
+    //!   ETag: "12345"
+    //!   ETag: W/"12345"
+    //!
+    //! The name of the header is matched in a case-insensitive fashion. This
+    //! function will return an empty string for non-standard, malformed, and
+    //! non-etag headers. It removes the quotes from the output and ignores a
+    //! space after the colon that separates the header name from the value.
+    //!
+    //! \param header Entire HTTP header field that includes the name and value.
+    //!
+    //! \return The parsed etag value or an empty string if the supplied header
+    //! contains no standard etag content.
+    //!
+    std::string ParseEtag(const std::string& header)
+    {
+        if (header.size() <= 8 || header[4] != ':') {
+            return std::string();
+        }
```

Comment:
Are we sure that etag: will always start at pos = 0? Is it possible that a webserver could put a space before the header field name? That was one of the reasons I did it the way I did it. Mine was not position dependent.

Reply:
Corollary, does the : always start at 4?

Reply:
Also, does there have to be a space between the colon and the opening quote? I don't see that specified in the standard, although it seems to be that way.

Reply:
If the web server is well-behaved, both of these will be true. An HTTP header name must not occur after whitespace. If it does, it is considered an extension of the content of a multi-line header value. A header name must not contain whitespace, so the colon must occur adjacent to the last non-whitespace character of the name.

Reply:
A minimum header size of 8 would allow empty etags to leak through if there are those crazy extra CRs, although it would be caught farther down. Better than mine, which processes every line.

Reply:
No, there does not need to be a space between the colon and the opening quote.

Reply:
If it's an empty etag, it comes back as an empty string just as the invalid etags do 🙂

Reply:
Yep. I think I am ok with it.